### PR TITLE
Fix RegEx for gcc > 4.9

### PR DESCRIPTION
**Pull request by HTPC-Schrauber**
On linux distributions with gcc > 4.9 cappuccino does not build because of the format of the RegEx.
I replaced all RegEx by a construct like new RegExp("xxx")
In the RegEx itself the \ had to be escaped for this to work.

This Pull Request refers to this [issue](https://github.com/cappuccino/cappuccino/issues/2499)

### CODE REVIEW

File: AppKit/CPTextView/CPTextView.j [hunk 1]
```diff
@@ -1848,18 +1848,18 @@ var kDelegateRespondsTo_textShouldBeginEditing
 
 + (JSObject)_wordBoundaryRegex
 {
-    return /(^[0-9][\.,])|(^.[^-\.,+#'"!§$%&/\(<\[\]>\)=?`´*\s{}\|¶])/m;
+    return new RegExp("(^[0-9][\\.,])|(^.[^-\\.,+#'\\\"!§$%&/\\(<\\[\\]>\\)=?`´*\\s{}\\|¶])", "m");
```
Comment (daboe01) [inline AppKit/CPTextView/CPTextView.j]: i am a bit uneasy about the \\\" (triple backslash)
should't this be \" (one backslash for the JS parser, not the regex parser)?
not 100% sure as i did not test this, though.

Comment (HTPC-Schrauber) [inline AppKit/CPTextView/CPTextView.j]: You are right. The triple backslashes are mistaken. No idea why I did this.
A single backslash is needed to escape the doublequote.

**Comment by daboe01** [comment]
-#new
Tools
+#needs-review

**Comment by daboe01** [comment]
+tools

**Comment by cacaodev** [comment]
@mrcarlberg do you think we can merge this ?

**Comment by mrcarlberg** [comment]
Merged! Thanks!

**Pull request merged**